src/types/colorTable.cpp

```diff
@@ -0,0 +1,1045 @@
+    til::color{ 0xD7, 0x87, 0xFF },
+    til::color{ 0xD7, 0xAF, 0x00 },
+    til::color{ 0xD7, 0xAF, 0x5F },
+    til::color{ 0xD7, 0xAF, 0x87 },
+    til::color{ 0xD7, 0xAF, 0xAF },
+    til::color{ 0xD7, 0xAF, 0xD7 },
+    til::color{ 0xD7, 0xAF, 0xFF },
+    til::color{ 0xD7, 0xD7, 0x00 },
+    til::color{ 0xD7, 0xD7, 0x5F },
+    til::color{ 0xD7, 0xD7, 0x87 },
+    til::color{ 0xD7, 0xD7, 0xAF },
+    til::color{ 0xD7, 0xD7, 0xD7 },
+    til::color{ 0xD7, 0xD7, 0xFF },
+    til::color{ 0xD7, 0xFF, 0x00 },
+    til::color{ 0xD7, 0xFF, 0x5F },
+    til::color{ 0xD7, 0xFF, 0x87 },
+    til::color{ 0xD7, 0xFF, 0xAF },
+    til::color{ 0xD7, 0xFF, 0xD7 },
+    til::color{ 0xD7, 0xFF, 0xFF },
+    til::color{ 0xFF, 0x00, 0x00 },
+    til::color{ 0xFF, 0x00, 0x5F },
+    til::color{ 0xFF, 0x00, 0x87 },
+    til::color{ 0xFF, 0x00, 0xAF },
+    til::color{ 0xFF, 0x00, 0xD7 },
+    til::color{ 0xFF, 0x00, 0xFF },
+    til::color{ 0xFF, 0x5F, 0x00 },
+    til::color{ 0xFF, 0x5F, 0x5F },
+    til::color{ 0xFF, 0x5F, 0x87 },
+    til::color{ 0xFF, 0x5F, 0xAF },
+    til::color{ 0xFF, 0x5F, 0xD7 },
+    til::color{ 0xFF, 0x5F, 0xFF },
+    til::color{ 0xFF, 0x87, 0x00 },
+    til::color{ 0xFF, 0x87, 0x5F },
+    til::color{ 0xFF, 0x87, 0x87 },
+    til::color{ 0xFF, 0x87, 0xAF },
+    til::color{ 0xFF, 0x87, 0xD7 },
+    til::color{ 0xFF, 0x87, 0xFF },
+    til::color{ 0xFF, 0xAF, 0x00 },
+    til::color{ 0xFF, 0xAF, 0x5F },
+    til::color{ 0xFF, 0xAF, 0x87 },
+    til::color{ 0xFF, 0xAF, 0xAF },
+    til::color{ 0xFF, 0xAF, 0xD7 },
+    til::color{ 0xFF, 0xAF, 0xFF },
+    til::color{ 0xFF, 0xD7, 0x00 },
+    til::color{ 0xFF, 0xD7, 0x5F },
+    til::color{ 0xFF, 0xD7, 0x87 },
+    til::color{ 0xFF, 0xD7, 0xAF },
+    til::color{ 0xFF, 0xD7, 0xD7 },
+    til::color{ 0xFF, 0xD7, 0xFF },
+    til::color{ 0xFF, 0xFF, 0x00 },
+    til::color{ 0xFF, 0xFF, 0x5F },
+    til::color{ 0xFF, 0xFF, 0x87 },
+    til::color{ 0xFF, 0xFF, 0xAF },
+    til::color{ 0xFF, 0xFF, 0xD7 },
+    til::color{ 0xFF, 0xFF, 0xFF },
+    til::color{ 0x08, 0x08, 0x08 },
+    til::color{ 0x12, 0x12, 0x12 },
+    til::color{ 0x1C, 0x1C, 0x1C },
+    til::color{ 0x26, 0x26, 0x26 },
+    til::color{ 0x30, 0x30, 0x30 },
+    til::color{ 0x3A, 0x3A, 0x3A },
+    til::color{ 0x44, 0x44, 0x44 },
+    til::color{ 0x4E, 0x4E, 0x4E },
+    til::color{ 0x58, 0x58, 0x58 },
+    til::color{ 0x62, 0x62, 0x62 },
+    til::color{ 0x6C, 0x6C, 0x6C },
+    til::color{ 0x76, 0x76, 0x76 },
+    til::color{ 0x80, 0x80, 0x80 },
+    til::color{ 0x8A, 0x8A, 0x8A },
+    til::color{ 0x94, 0x94, 0x94 },
+    til::color{ 0x9E, 0x9E, 0x9E },
+    til::color{ 0xA8, 0xA8, 0xA8 },
+    til::color{ 0xB2, 0xB2, 0xB2 },
+    til::color{ 0xBC, 0xBC, 0xBC },
+    til::color{ 0xC6, 0xC6, 0xC6 },
+    til::color{ 0xD0, 0xD0, 0xD0 },
+    til::color{ 0xDA, 0xDA, 0xDA },
+    til::color{ 0xE4, 0xE4, 0xE4 },
+    til::color{ 0xEE, 0xEE, 0xEE },
+};
+
+static til::static_map xorgAppColorTable{

```

Comment:
Now that #7640 has landed, can you switch this to a presorted_static_map, make it `static constexpr`, and make sure the keys are sorted in the natural order that string_view uses?

to save space in the binary, is there a way we can store them as `""sv` instead of `L""sv`? x11 guarantees that these are Latin1, so we can use a shortcutted version of string conversion that fails if any characters `>` 127 are present, and deletes spaces, and lowercases at the same time etc etc

Reply:
By natural order do you mean:

```plaintext
gray0
gray1
gray2
```
over

```plaintext
gray0
gray1
gray10
gray100
gray11
```

Reply:
As long as it matches the order you get from sorting a vector of `std::string_view`s!

Reply:
(I meant the "natural order" for the key type, specifically. Which means largely "whatever a normal sort function would turn out")

Reply:
it looks like the order you are using is correct, based on my experimentation.

Reply:
Yeah it is sorted programmatically using vscode, I believe.